sktime/numeric/utils.py

```diff
@@ -97,3 +97,20 @@ def allclose_sparse(A, B, rtol=1e-5, atol=1e-8):
     else:
         return False
 
+
+def drop_nan_rows(x, x_dot):
+    r"""
```

Comment:
I believe in terms or reusability this could have a signature like `drop_nan_rows(arr, *arrays)` so that all arr-nan rows are dropped in arr and all arrays.